lib/Stats.js

```diff
@@ -108,16 +119,8 @@ class Stats {
 		const showWarnings = optionOrLocalFallback(options.warnings, true);
 		const warningsFilter = optionOrFallback(options.warningsFilter, null);
 		const showPublicPath = optionOrLocalFallback(options.publicPath, !forToString);
-		const excludeModules = [].concat(optionOrFallback(options.exclude, [])).map(item => {
-			if(typeof item === "string") {
-				const regExp = new RegExp(`[\\\\/]${item.replace(/[\-\[\]\/\{\}\(\)\*\+\?\.\\\^\$\|]/g, "\\$&")}([\\\\/]|$|!|\\?)`); // eslint-disable-line no-useless-escape
-				return ident => regExp.test(ident);
-			}
-			if(item && typeof item === "object" && typeof item.test === "function")
-				return ident => item.test(ident);
-			if(typeof item === "function")
-				return item;
-		});
+		const excludeModules = [].concat(optionOrFallback(options.exclude, [])).map(testAgainstGivenOption);
```

Comment:
Could you rename this to `excludeModules` with backward compatible fallback to `exclude`?

`[].concat(optionOrFallback(options.excludeModules, optionOrFallback(options.exclude, [])))`

or refactor `optionOrFallback` to take any number of arguments...